Sources/TuistKit/Commands/TuistCommand.swift

```diff
@@ -248,6 +262,16 @@ public struct TuistCommand: AsyncParsableCommand {
 
     static func processArguments(_ arguments: [String]? = nil) -> [String]? {
         let arguments = arguments ?? Array(ProcessInfo.processInfo.arguments)
-        return arguments.filter { $0 != "--verbose" && $0 != "--quiet" }
+        return arguments.filter { $0 != "--verbose" && $0 != "--quiet" && $0 != "--warnings-as-errors" }
+    }
+
+    private struct WarningsAsErrorsError: FatalError {
+        var description: String {
+            "The command completed successfully, but there were warnings."
+        }
+
+        var type: ErrorType {
+            .abort
+        }
```

Comment:
Let's use `LocalizedError` which is our new convention. `FatalError` is a deprecated pattern.

```suggestion
    private struct WarningsAsErrorsError: LocalizedError {
        var errorDescription: String? {
            "The command completed successfully, but there were warnings."
        }
```